core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -257,6 +264,37 @@ class ReplicaManager(val config: KafkaConfig,
     }
   }
 
+  // Changes are initially deferred when using a Raft-based metadata quorum, and they may flip-flop to not
+  // being deferred and being deferred again thereafter as the broker (re)acquires/loses its lease.
+  // Changes are never deferred when using ZooKeeper.  When true, this indicates that we should transition
+  // online partitions to the deferred state if we see a metadata update for that partition.
+  @volatile private var deferringMetadataChanges: Boolean = !config.requiresZookeeper
```

Comment:
Why does this need to volatile?

Reply:
Good point.  The same thread will be both managing this value and submitting metadata changes, so it can remember if it has already started deferring changes if it needs to do so; plus everything here related to deferral happens within the lock anyway.  I removed `@volatile`.